packages/flutter/lib/src/material/menu_bar_theme.dart

```diff
@@ -0,0 +1,333 @@
+      itemPadding: itemPadding ?? this.itemPadding,
+      itemShape: itemShape ?? this.itemShape,
+    );
+  }
+
+  /// Linearly interpolate between two [MenuBarThemeData]s.
+  ///
+  /// If both arguments are null, then null is returned.
+  ///
+  /// {@macro dart.ui.shadow.lerp}
+  static MenuBarThemeData? lerp(MenuBarThemeData? a, MenuBarThemeData? b, double t) {
+    assert(t != null);
+    if (a == null && b == null) {
+      return null;
+    }
+    return MenuBarThemeData(
+      barHeight: lerpDouble(a?.barHeight, b?.barHeight, t),
+      barPadding: EdgeInsets.lerp(a?.barPadding, b?.barPadding, t),
+      barBackgroundColor: _lerpProperties<Color?>(a?.barBackgroundColor, b?.barBackgroundColor, t, Color.lerp),
+      barElevation: _lerpProperties<double?>(a?.barElevation, b?.barElevation, t, lerpDouble),
+      menuBackgroundColor: _lerpProperties<Color?>(a?.menuBackgroundColor, b?.menuBackgroundColor, t, Color.lerp),
+      menuElevation: _lerpProperties<double?>(a?.menuElevation, b?.menuElevation, t, lerpDouble),
+      menuShape: _lerpProperties<ShapeBorder?>(a?.menuShape, b?.menuShape, t, ShapeBorder.lerp),
+      menuPadding: EdgeInsets.lerp(a?.menuPadding, b?.menuPadding, t),
+      itemBackgroundColor: _lerpProperties<Color?>(a?.itemBackgroundColor, b?.itemBackgroundColor, t, Color.lerp),
+      itemForegroundColor: _lerpProperties<Color?>(a?.itemForegroundColor, b?.itemForegroundColor, t, Color.lerp),
+      itemOverlayColor: _lerpProperties<Color?>(a?.itemOverlayColor, b?.itemOverlayColor, t, Color.lerp),
+      itemTextStyle: _lerpProperties<TextStyle?>(a?.itemTextStyle, b?.itemTextStyle, t, TextStyle.lerp),
+      itemPadding: EdgeInsets.lerp(a?.itemPadding, b?.itemPadding, t),
+      itemShape: _lerpProperties<OutlinedBorder?>(
+        a?.itemShape,
+        b?.itemShape,
+        t,
+        (ShapeBorder? a, ShapeBorder? b, double t) {
+          return ShapeBorder.lerp(a, b, t) as OutlinedBorder?;
+        },
+      ),
+    );
+  }
+
+  static MaterialStateProperty<T>? _lerpProperties<T>(
+    MaterialStateProperty<T>? a,
+    MaterialStateProperty<T>? b,
+    double t,
+    T Function(T?, T?, double) lerpFunction,
+  ) {
+    // Avoid creating a _LerpProperties object for a common case.
+    if (a == null && b == null) {
+      return null;
+    }
+    return _LerpProperties<T>(a, b, t, lerpFunction);
+  }
+
+  @override
+  int get hashCode {
+    return Object.hash(
+      barHeight,
+      barPadding,
+      barBackgroundColor,
+      barElevation,
+      menuBackgroundColor,
+      menuElevation,
+      menuShape,
+      menuPadding,
+      itemBackgroundColor,
+      itemForegroundColor,
+      itemOverlayColor,
+      itemTextStyle,
+      itemPadding,
+      itemShape,
+    );
+  }
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other)) {
+      return true;
+    }
+    if (other.runtimeType != runtimeType) {
+      return false;
+    }
+    return other is MenuBarThemeData &&
```

Comment:
Conventional formatting for this kind of thing is like:

```dart
    return other is ChipThemeData
        && other.backgroundColor == backgroundColor
        && other.deleteIconColor == deleteIconColor
        ...
```

Reply:
Yeah, I know.  Once the changes settle down a bit I'll make a formatting pass to normalize the formatting before submitting.